src/lib/desktop.ts

```diff
@@ -695,14 +755,138 @@ export const readTokenFile = async () => {
+  if (window.electron.exists(path)) {
+    const configurationJSON: string = await window.electron.readFile(path, {
+      encoding: 'utf-8',
+    })
+    if (!configurationJSON) return null
+    return JSON.parse(configurationJSON)
+  }
+  return null
+}
+
+export const writeEnvironmentConfigurationToken = async (
+  environmentName: string,
+  token: string
+) => {
+  environmentName = environmentName.trim()
+  const path = await getEnvironmentConfigurationPath(environmentName)
+  const environmentConfiguration =
+    await getEnvironmentConfigurationObject(environmentName)
+  environmentConfiguration.token = token
+  const requestedConfiguration = JSON.stringify(environmentConfiguration)
+  const result = window.electron.writeFile(path, requestedConfiguration)
+  console.log(`wrote ${environmentName}.json to disk`)
   return result
 }
 
+export const writeEnvironmentConfigurationPool = async (
+  environmentName: string,
+  pool: string
+) => {
+  pool = pool.trim()
+  const path = await getEnvironmentConfigurationPath(environmentName)
+  const environmentConfiguration =
+    await getEnvironmentConfigurationObject(environmentName)
+  environmentConfiguration.pool = pool
+  const requestedConfiguration = JSON.stringify(environmentConfiguration)
+  const result = window.electron.writeFile(path, requestedConfiguration)
+  console.log(`wrote ${environmentName}.json to disk`)
+  return result
+}
+
+export const getEnvironmentConfigurationObject = async (
+  environmentName: string
+) => {
+  let environmentConfiguration =
+    await readEnvironmentConfigurationFile(environmentName)
+  if (environmentConfiguration === null) {
+    const initialConfiguration: EnvironmentConfiguration = {
+      token: '',
+      pool: '',
+      domain: environmentName,
+    }
+    environmentConfiguration = initialConfiguration
+  }
+  return environmentConfiguration
+}
+
+export const readEnvironmentConfigurationPool = async (
+  environmentName: string
+) => {
+  const environmentConfiguration =
+    await readEnvironmentConfigurationFile(environmentName)
+  if (!environmentConfiguration?.pool) return ''
+  return environmentConfiguration.pool.trim()
+}
+
+export const readEnvironmentConfigurationToken = async (
+  environmentName: string
+) => {
+  const environmentConfiguration =
+    await readEnvironmentConfigurationFile(environmentName)
+  if (!environmentConfiguration?.token) return ''
+  return environmentConfiguration.token.trim()
+}
+
+export const readEnvironmentFile = async () => {
+  let environmentFilePath = await getEnvironmentFilePath()
+
+  if (window.electron.exists(environmentFilePath)) {
+    const environment: string = await window.electron.readFile(
+      environmentFilePath,
+      {
+        encoding: 'utf-8',
+      }
+    )
+    if (!environment) return ''
+    return environment.trim()
+  }
+  return ''
+}
+
+/**
+ * We store the environment file on disk because we store the token.txt across app installs
```

Comment:
This seems to be the only reference to `token.txt` in this PR. Is that still the correct filename?